wagtail_streamfield_migration_toolkit/utils.py

```diff
@@ -1,6 +1,253 @@
+from wagtail.blocks import ListBlock, StreamBlock, StructBlock
 
+from wagtail_streamfield_migration_toolkit.operations import BaseBlockOperation
 
-def apply_changes_to_raw_data(raw_data, block_path, operation, *args, **kwargs):
-    altered_raw_data = raw_data
-    # Logic will be here 
-    return altered_raw_data
\ No newline at end of file
+
+# TODO handle block_defs not existing? we'll do this later
+
+
+def is_block_at_path_start(block_name, block_path):
+    return block_name == block_path[0]
+
+
+def map_block_value(
+    block_value, block_def, block_path, operation: BaseBlockOperation, **kwargs
+):
+    """
+    Maps the value of a block. 
+
+    Args:
+        block_value:
+            The value of the block. This would be a list or dict of children for structural blocks.
+        block_def:
+            The definition of the block.
+        block_path:
+            A '.' separated list of names of the blocks from the current block (not included) to
+            the nested block of which the value will be passed to the operation.
+        operation:
+            An Operation class instance (extends `BaseBlockOperation`), which has an `apply` method
+            for mapping values.
+
+    Returns:
+        mapped_value:
+    """
+
+    # If the `block_path` length is 0, that means we've reached the end of the block path, that
+    # is, the block where we need to apply the operation. Note that we are asking the user to
+    # pass "item" as part of the block path for list children, so it won't give rise to any
+    # problems here.
+    if len(block_path) == 0:
+        mapped_value = operation.apply(block_value)
+        return mapped_value
```

Comment:
```suggestion
        return operation.apply(block_value)
```

Up to you, I would do this in one line